qCC/ccComparisonDlg.cpp

```diff
@@ -129,24 +129,20 @@ ccComparisonDlg::ccComparisonDlg(	ccHObject* compEntity,
 		filterVisibilityCheckBox->setEnabled(m_refCloud && m_refCloud->isA(CC_TYPES::POINT_CLOUD) && static_cast<ccPointCloud*>(m_refCloud)->hasSensor());
 	}
 
-	connect(cancelButton,			&QPushButton::clicked,					this,	ccComparisonDlg::cancelAndExit);
-	connect(okButton,				&QPushButton::clicked,					this,	ccComparisonDlg::applyAndExit);
-	connect(computeButton,			&QPushButton::clicked,					this,	ccComparisonDlg::computeDistances);
-	connect(histoButton,			&QPushButton::clicked,					this,	ccComparisonDlg::showHisto);
-	connect(maxDistCheckBox,		&QCheckBox::toggled,					this,	ccComparisonDlg::maxDistUpdated);
-	connect(localModelComboBox, static_cast<void (QComboBox::*)(int)> (&QComboBox::currentIndexChanged),		this,	ccComparisonDlg::locaModelChanged);
-	connect(maxSearchDistSpinBox, static_cast<void (QDoubleSpinBox::*)(double)> (&QDoubleSpinBox::valueChanged),this,	ccComparisonDlg::maxDistUpdated);
+	connect(cancelButton,			&QPushButton::clicked,					this,	&ccComparisonDlg::cancelAndExit);
+	connect(okButton,				&QPushButton::clicked,					this,	&ccComparisonDlg::applyAndExit);
+	connect(computeButton,			&QPushButton::clicked,					this,	&ccComparisonDlg::computeDistances);
+	connect(histoButton,			&QPushButton::clicked,					this,	&ccComparisonDlg::showHisto);
+	connect(maxDistCheckBox,		&QCheckBox::toggled,					this,	&ccComparisonDlg::maxDistUpdated);
+	connect(localModelComboBox, static_cast<void (QComboBox::*)(int)> (&QComboBox::currentIndexChanged),		this,	&ccComparisonDlg::locaModelChanged);
+	connect(maxSearchDistSpinBox, static_cast<void (QDoubleSpinBox::*)(double)> (&QDoubleSpinBox::valueChanged),this,	&ccComparisonDlg::maxDistUpdated);
 
 	//be sure to show the dialog before computing the approx distances
 	//(otherwise the progress bars appear anywhere!)
-	if (!m_noDisplay)
-	{
-		show();
-		QCoreApplication::processEvents();
-	}
-
-	//compute approximate results and unlock GUI
-	computeApproxDistances();
+	QTimer::singleShot(0, this, [=]() { //singleshot 0 time will fire once after all events in the queue are processed
```

Comment:
Interesting solution

Reply:
Do we still need to call 'processEvents' then?

Reply:
I'm not totally sure, I only tested removing show(); and removing the check for m_noDisplay (each area this is used calls show like this)

```C++
m_compDlg = new ccComparisonDlg(compCloud, refCloud, ccComparisonDlg::CLOUDCLOUD_DIST, this);
	connect(m_compDlg, &QDialog::finished, this, &MainWindow::deactivateComparisonMode);
	m_compDlg->show();
```
I found that after removing the show function I had to add the single shot to avoid the progress bars issue. I'll run a quick test to see if it makes any difference for me

Reply:
I couldn't tell any difference. with or without 'processEvents' so I believe it is no longer required